mllib/src/main/scala/org/apache/spark/ml/feature/IDF.scala

```diff
@@ -178,10 +187,10 @@ object IDFModel extends MLReadable[IDFModel] {
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
       val dataPath = new Path(path, "data").toString
       val data = sparkSession.read.parquet(dataPath)
-      val Row(idf: Vector) = MLUtils.convertVectorColumnsToML(data, "idf")
-        .select("idf")
-        .head()
-      val model = new IDFModel(metadata.uid, new feature.IDFModel(OldVectors.fromML(idf)))
+      val Row(idf: Vector, df: Seq[_], numDocs: Long) =
```

Comment:
Unfortunately this is going to be a bit more complex, as it has to handle the case of older models saved without this info. See how some other implementations of save/load use versions to handle this.

Reply:
I have edited it now to add support for older models. Apologies, missed it previously.

Should a test case be added to test for older models? I check PCASuite, FPGrowth and KMeanSuite where the models were changed with version. Didn't find test to check for older models. Think it is a norm. Can add if needed.

Squashed the commits.